sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/ResourceManagerThrottlingPolicy.java

```diff
@@ -7,86 +7,28 @@
 import com.azure.core.http.HttpPipelineNextPolicy;
 import com.azure.core.http.HttpResponse;
 import com.azure.core.http.policy.HttpPipelinePolicy;
-import com.azure.core.util.DateTimeRfc1123;
-import com.azure.core.util.FluxUtil;
-import com.azure.core.util.logging.ClientLogger;
-import com.azure.resourcemanager.resources.fluentcore.utils.SdkContext;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerThrottlingInfo;
 import reactor.core.publisher.Mono;
 
-import java.nio.charset.StandardCharsets;
-import java.time.Duration;
-import java.time.Instant;
-import java.time.OffsetDateTime;
-import java.util.concurrent.TimeUnit;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.function.Consumer;
 
 /**
- * A Http Pipeline Policy for automatic retry when Azure Resource Manager is throttling
- * because of too many read/write requests.
- * <p>
- * For each subscription and tenant, Azure Resource Manager limits read requests to 15,000 per hour and
- *   write requests to 1,200 per hour. These limits apply to each Azure Resource Manager instance.
+ * A Http Pipeline Policy for automatic send throttling rate limit info to a call back function
  */
 public class ResourceManagerThrottlingPolicy implements HttpPipelinePolicy {
-    private static final String RETRY_MESSAGE_FORMAT =
-        "Azure Resource Manager read/write per hour limit reached. Will retry in: %d seconds";
+    private final Consumer<? super ResourceManagerThrottlingInfo> callback;
```

Comment:
The callback seems better to take a `HttpResponse`, so received be able to match request/response with throttling info.

Reply:
done